generator/resources/ec2_userdata_test_matrix.json

```diff
@@ -0,0 +1,13 @@
+[
+    {
+        "os": "ol9",
```

Comment:
Since the this PR is specific to the migration of Linux test based on the internal integration test, supporting Windows userdata test is out of scope. However I do agree there should be more similar testing done on Windows as well.